src/main/java/uk/gov/digital/justice/hmpps/sentenceplan/jpa/entity/StepEntity.java

```diff
@@ -24,13 +27,53 @@
 
     public StepEntity(StepOwner owner, String ownerOther, String description, String strength, StepStatus status, List<UUID> needs, String intervention) {
 
+        updateStep(owner, ownerOther, description, strength, status, needs, intervention);
         this.id = UUID.randomUUID();
+    }
+
+    public void updateStep(StepOwner owner, String ownerOther, String description, String strength, StepStatus status, List<UUID> needs, String intervention) {
+
+        validateNeeds(needs);
+        validateOwner(owner, ownerOther);
+        validateDescription(description, intervention);
+
         this.owner = owner;
         this.ownerOther = ownerOther;
-        this.description = description;
+
+        // Overwrite the description
+        if(StringUtils.isEmpty(intervention)) {
+            this.description = description;
+            this.intervention = intervention;
+        } else {
+            this.description = intervention;
+            this.intervention = intervention;
```

Comment:
Duplication:
`this.intervention = intervention;`
Maybe set description with a tertiary.

Reply:
I have fixed this in my current branch as I am working in this area again.